src/modules/exampleEntity/application/exampleEntity.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable, Inject, ConflictException } from "@nestjs/common";
+import { IExampleEntityRepository, EXAMPLE_ENTITY_REPOSITORY_TOKEN } from "../domain/exampleEntity-repository.interface";
+import { CreateExampleEntityDTO } from "./create-exampleEntity.dto";
+import { ExampleEntity } from "../domain/exampleEntity.entity";
+
+@Injectable()
+export class ExampleEntityService{
+    @Inject(EXAMPLE_ENTITY_REPOSITORY_TOKEN)
+    private readonly exampleEntityRepository: IExampleEntityRepository;
+    async createExampleentity(exampleEntityDto: CreateExampleEntityDTO):Promise<ExampleEntity>{
+        if(await this.exampleEntityRepository.findByEmail(exampleEntityDto.email)){
+            throw new ConflictException("the provided email already was used");
```

Comment:
Acho que fica mais claro e padrão, mas é só example entity então não precisa alterar 
```suggestion
            throw new ConflictException("the provided email is already in use");
```

Reply:
De acordo. Alterado localmente.